bpf/lib/nodeport_egress.h

```diff
@@ -123,6 +127,7 @@ static __always_inline int nodeport_snat_fwd_ipv6(struct __ctx_buff *ctx,
 	return ret;
 }
 
+#if !defined(NODEPORT_EGRESS_REVDNAT_ONLY)
```

Comment:
We did quite a bit of work to get rid of conditionally compiling tail calls in https://github.com/cilium/cilium/pull/30467.

Just compiling out the call site should ensure we do not load it due to some loader smarts. Is that good enough?

Reply:
Right. It just really sucks to fix it in this way because it only fuels our addiction to per-processor macros. Plus, the next time someone modifies anything it might be the `bpf_host.c` or `bpf_overlay.c` path as well that shits the bed. It feels like hiding mines.

Reply:
It seems like its also somewhat dependent on clang version ect. I am not able to repro this at all for example. After checking out this branch and commenting out the `#define NODEPORT_EGRESS_REVDNAT_ONLY` and building the bpf_wireguard.o, even with the example you provided out of bounds I only get up to 488 bytes of stack usage (as reported with `-fstack-usage`).